Fix failing ExtEvDataServiceSpec

```
timeout (3 seconds) during expectMessage while waiting for Completion(Actor[pekko://ExtEvDataServiceSpec/user/$y/$$a-adapter#-2101030711],None)
java.lang.AssertionError: timeout (3 seconds) during expectMessage while waiting for Completion(Actor[pekko://ExtEvDataServiceSpec/user/$y/$$a-adapter#-2101030711],None)
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -223,6 +223,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - Fix schedule lock in PrimaryServiceProxy [#1289](https://github.com/ie3-institute/simona/issues/1289)
 - Fix vn_simona time series mapping [#1288](https://github.com/ie3-institute/simona/issues/1288)
 - Fix hierarchical em structures leading to exception while initializing [#1293](https://github.com/ie3-institute/simona/issues/1293)
+- Fix failing `ExtEvDataServiceSpec` [#1297](https://github.com/ie3-institute/simona/issues/1297)
 
 ### Removed
 - Removed `SimonaListerner` and related code [#1205](https://github.com/ie3-institute/simona/issues/1205)

--- a/src/main/scala/edu/ie3/simona/service/ExtDataSupport.scala
+++ b/src/main/scala/edu/ie3/simona/service/ExtDataSupport.scala
@@ -14,7 +14,7 @@ import edu.ie3.simona.ontology.messages.services.ServiceMessage.{
   WrappedExternalMessage,
 }
 import edu.ie3.simona.service.ServiceStateData.ServiceConstantStateData
-import org.apache.pekko.actor.typed.scaladsl.Behaviors
+import org.apache.pekko.actor.typed.scaladsl.{ActorContext, Behaviors}
 import org.apache.pekko.actor.typed.{ActorRef, Behavior}
 
 /** Trait that enables handling of external data.
@@ -46,10 +46,10 @@ trait ExtDataSupport[
         Behaviors.same
     }
 
-  override private[service] def idleExternal(implicit
+  override protected def idleExternal(implicit
       stateData: S,
       constantData: ServiceConstantStateData,
-  ): Behavior[T] = Behaviors.receive {
+  ): PartialFunction[(ActorContext[T], T), Behavior[T]] = {
     case (_, WrappedExternalMessage(extMsg)) =>
       val updatedStateData = handleDataMessage(extMsg)(stateData)
 
@@ -60,13 +60,6 @@ trait ExtDataSupport[
         handleDataResponseMessage(extResponseMsg)(stateData)
 
       idle(updatedStateData, constantData)
-
-    case (ctx, unsupported) =>
-      ctx.log.debug(
-        s"Received unsupported message: $unsupported! Message forwarded to be handled internally."
-      )
-      ctx.self ! unsupported
-      idleInternal
   }
 
   /** Handle a message from outside the simulation

--- a/src/main/scala/edu/ie3/simona/service/TypedSimonaService.scala
+++ b/src/main/scala/edu/ie3/simona/service/TypedSimonaService.scala
@@ -161,12 +161,16 @@ abstract class TypedSimonaService[
   final protected def idle(implicit
       stateData: S,
       constantData: ServiceConstantStateData,
-  ): Behavior[T] = idleExternal
+  ): Behavior[T] = Behaviors.receive[T] { case (ctx, msg) =>
+    idleInternal
+      .orElse(idleExternal)
+      .applyOrElse((ctx, msg), unhandled.tupled)
+  }
 
-  private[service] def idleInternal(implicit
+  private def idleInternal(implicit
       stateData: S,
       constantData: ServiceConstantStateData,
-  ): Behavior[T] = Behaviors.receive {
+  ): PartialFunction[(ActorContext[T], T), Behavior[T]] = {
     // agent registration process
     case (ctx, registrationMsg: ServiceRegistrationMessage) =>
       /* Someone asks to register for information from the service */
@@ -207,26 +211,27 @@ abstract class TypedSimonaService[
       )
 
       idle(updatedStateData, constantData)
+  }
 
-    // unhandled message
-    case (ctx, x) =>
-      ctx.log.error("Unhandled message received:{}", x)
+  private def unhandled: (ActorContext[T], T) => Behavior[T] = {
+    case (ctx, msg) =>
+      ctx.log.error("Unhandled message received:{}", msg)
       Behaviors.unhandled
   }
 
   /** Internal api method that allows handling incoming messages from external
-    * simulations
+    * simulations.
     *
     * @param stateData
-    *   the state data of this service
+    *   The state data of this service.
     * @return
-    *   the [[idleExternal()]] behavior as default, to override extend
-    *   [[ExtDataSupport]]
+    *   Empty partial function as default. To override, extend
+    *   [[ExtDataSupport]].
     */
-  private[service] def idleExternal(implicit
+  protected def idleExternal(implicit
       stateData: S,
       constantData: ServiceConstantStateData,
-  ): Behavior[T] = idleInternal
+  ): PartialFunction[(ActorContext[T], T), Behavior[T]] = PartialFunction.empty
 
   /** Initialize the concrete service implementation using the provided
     * initialization data. This method should perform all heavyweight tasks

